fe/fe-core/src/main/java/com/starrocks/analysis/SlotDescriptor.java

```diff
@@ -164,6 +165,8 @@ public void setColumn(Column column) {
                         scalarType.getPrimitiveType(),
                         scalarType.getScalarPrecision(),
                         scalarType.getScalarScale());
+            } else if (this.originType.isVarchar() && FeConstants.setLengthForVarchar) {
+                this.type = ScalarType.createVarcharType(scalarType.getLength());
```

Comment:
why need to handle varchar in a particular way?  how about the default varchar type or  just the clone of the column.getType()?

Reply:
> why need to handle varchar in a particular way? how about the default varchar type or just the clone of the column.getType()?

because I want if this.type is varchar, its length is not -1 but originType.length.  FeConstants.setLengthForVarchar is used for FE UT. There are too many UT using explain's output, so I have to modify too many UT. So I add this config for some UT